providers/hubspot/subscriptionEvent.go

```diff
@@ -18,13 +19,22 @@ type SubscriptionEvent struct {
 	EventId          int    `json:"eventId"`
 	SubscriptionId   int    `json:"subscriptionId"`
 	PortalId         int    `json:"portalId"`
-	OccurredAt       int    `json:"occurredAt"`
+	OccurredAt       int    `json:"occurredAt"` // in milliseconds
 	SubscriptionType string `json:"subscriptionType"`
 	AttemptNumber    int    `json:"attemptNumber"`
-	ObjectId         int    `json:"objectId"`
 	ChangeSource     string `json:"changeSource"`
-	PropertyName     string `json:"propertyName"`
-	PropertyValue    string `json:"propertyValue"`
+	// Optional fields
+	ObjectId   *int    `json:"objectId,omitempty"`
+	ChangeFlag *string `json:"changeFlag,omitempty"`
```

Comment:
can you turn these `*string` types back into `string`? References to strings are annoying to deal with, since you have to check that it's not nil before dereferencing and it doesn't convey any additional information than a simple string type. With a string type, you can easily check `if ChangeFlag === ""` to see if something is an empty string. (As opposed to something like to a bool value, where `*bool` is actually useful since it allows you to differentiate between an empty value and false).

I know we currently have a lot of `*string` in our codebase, because the openapi generator turns optional strings into `*string` by default, unless you add the `x-go-type-skip-optional-pointer: true` annotation. But we should actively try to avoid introducing more of it into the codebase.

Reply:
Never mind, this is a good approach since it allows us to more easily strip out empty fields when sending the webhook message to builders, see https://github.com/amp-labs/connectors/pull/1251#issuecomment-2566067227

Reply:
Omitempty will still omit empty strings when you JSON marshal. See https://go.dev/play/p/zU4Q3eohfi4, so `*string` is not necessary, `string` type will work.